cli/replicaset/cartridge.go

```diff
@@ -103,22 +126,30 @@ func (c *CartridgeApplication) GetReplicasets() (Replicasets, error) {
 		Orchestrator: OrchestratorCartridge,
 	}
 
-	var err error
-	if c.preferred != nil {
-		replicasets, err = NewCartridgeInstance(c.preferred).GetReplicasets()
-	} else {
-		err = EvalAny(c.runningCtx.Instances, InstanceEvalFunc(
-			func(_ running.InstanceCtx, evaler connector.Evaler) (bool, error) {
-				var err error
-				replicasets, err = NewCartridgeInstance(evaler).GetReplicasets()
+	var topology cartridgeTopology
+	err := EvalForeachAlive(c.runningCtx.Instances, InstanceEvalFunc(
+		func(inst running.InstanceCtx, evaler connector.Evaler) (bool, error) {
+			newTopology, err := getCartridgeTopology(evaler)
+			if err != nil {
 				return true, err
-			},
-		))
-	}
+			}
+			if topology.isBootstrapped {
+				if newTopology.isBootstrapped {
+					topology = newTopology
+				}
+			} else {
+				topology = newTopology
+			}
+
+			// Stop if we already found a valid topology.
+			return topology.isBootstrapped && !topology.IsCritical, nil
```

Comment:
Looks like, I've missed something. Why do make `topology = newTopology` and only after that check `IsCrtitical`. Do we really need to perform an assignment if new topology is not bootstrapped or has critical issues?

Reply:
If a bootstrapped instance is found, we never switch it to a non-bootstrapped. And If this instance hasn't critical issues - we stopping the search. But if every instance has critical on it - there's nothing left to us except to take topology from them.

Reply:
Okay.